dashboard/app/controllers/test_controller.rb

```diff
@@ -90,19 +90,25 @@ def assign_course_and_unit_as_student
     script = Unit.find_by_name(params.require(:script_name))
     course = UnitGroup.find_by_name(params.require(:course_name))
 
-    name = "Fake User"
-    email = "user#{Time.now.to_i}_#{rand(1_000_000)}@test.xx"
-    password = name + "password"
-    attributes = {
-      name: name,
-      email: email,
-      password: password,
-      user_type: "teacher",
-      age: "21+"
-    }
-    fake_user = User.create!(attributes)
+    teacher_name = params[:teacher_name]
+    teacher_user = nil
+    if teacher_name
```

Comment:
Can you say a bit more why you decided to modify this in this way?

Reply:
This adds a way to create the section for a student with a specified course and unit with a specific teacher. Previously it would always create the section under a new fake user teacher, which didn't allow me to sign in as that teacher and see the section.